source/common/api/posix/os_sys_calls_impl.cc

```diff
@@ -282,5 +283,53 @@ SysCallBoolResult OsSysCallsImpl::socketTcpInfo([[maybe_unused]] os_fd_t sockfd,
   return {false, EOPNOTSUPP};
 }
 
+bool OsSysCallsImpl::supportsGetifaddrs() const {
+// TODO: eliminate this branching by upstreaming an alternative Android implementation
+// e.g.: https://github.com/envoyproxy/envoy-mobile/blob/main/third_party/android/ifaddrs-android.h
+#if defined(__ANDROID_API__) && __ANDROID_API__ < 24
```

Comment:
@goaway I attempted to coerce the Socket options pattern here. The problem is that the definition of the macro in Envoy Mobile still wouldn't be available to this code. Unless I am missing some macro/preprocessor magic. Any pointers would be helpful.

Reply:
Fixed by adding a setter for a lambda in the singleton.